src/com/haskforce/highlighting/annotation/HaskellDocumentationProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.haskforce.highlighting.annotation;
+
+import com.google.common.collect.Lists;
+import com.haskforce.highlighting.annotation.external.GhcMod;
+import com.haskforce.highlighting.annotation.external.GhcModi;
+import com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator;
+import com.haskforce.highlighting.annotation.external.TypeInfoUtil;
+import com.haskforce.settings.ToolKey;
+import com.haskforce.utils.ExecUtil;
+import com.intellij.lang.documentation.AbstractDocumentationProvider;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.LogicalPosition;
+import com.intellij.openapi.editor.VisualPosition;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class HaskellDocumentationProvider extends AbstractDocumentationProvider {
```

Comment:
Never mind, reproduced, fixed. I performed a clean up that was a bit too harsh, and totally wrong. Anyways, should work now. Let me know.